skrub/_table_vectorizer.py

```diff
@@ -567,7 +567,16 @@ def _apply_cast(self, X: pd.DataFrame) -> pd.DataFrame:
                     categories=known_categories.union(new_categories)
                 )
                 self.types_[col] = dtype
-        X = X.astype(self.types_)
+        for col, dtype in self.types_.items():
+            # if the inferred dtype is numerical or datetime,
+            # we want to ignore entries that cannot be converted
+            # to this dtype
+            if pd.api.types.is_numeric_dtype(dtype):
+                X[col] = pd.to_numeric(X[col], errors="coerce")
```

Comment:
With the current usage of "coerce" it is hard to do a warning if an invalid value is encountered.

If we feel that it is important to warn (I am leaning in this direction but not 100% sure), one way to do it is to do a try/except with 'errors="raise"', if an error (the "except" block), capture it, pop up a warning (ideally informative) and then do the same call but with 'error="coerce"'.